python/cudf/cudf/core/column/string.py

```diff
@@ -5594,16 +5594,15 @@ def data(self):
         return self._data
 
     def all(self, skipna: bool = True) -> bool:
-        # The skipna argument is only used for numerical columns.
-        # If all entries are null the result is True, including when the column
-        # is empty.
-
+        if self.null_count == self.size:
+            return True
         raise NotImplementedError("`all` not implemented for `StringColumn`")
 
     def any(self, skipna: bool = True) -> bool:
-        # The skipna argument is only used for numerical columns.
-        # If all entries are null the result is True, including when the column
-        # is empty.
+        if not skipna and self.has_nulls():
+            return True
```

Comment:
If `skipna` is `False` and there are any `pd.NA` values, pandas 2.1.4 raises an error. This code would return `True`, from what I can tell. Do we need to raise an error?

```python
import pandas as pd
s = pd.Series([pd.NA, pd.NA])
print(s.any(skipna=True))  # False
print(s.any(skipna=False))  # TypeError: boolean value of NA is ambiguous
```

Reply:
Yes please!

Reply:
Fixed it.